options/internal/m68k-include/mlibc/thread.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <stdint.h>
+#include <mlibc/internal-sysdeps.hpp>
+#include <mlibc/tcb.hpp>
+#include <bits/ensure.h>
+
+namespace mlibc {
+
+extern "C" void *__m68k_read_tp();
+
+inline Tcb *get_current_tcb() {
+	// On m68k, the TCB is below 0x7000 below the thread pointer.
+	void *ptr = __m68k_read_tp();
+	return reinterpret_cast<Tcb *>((uintptr_t)ptr - 0x7000 - sizeof(Tcb));
```

Comment:
can you use tlsOffsetFromTp?

Reply:
It's defined elsewhere, so I followed what the aarch64 port does: https://github.com/managarm/mlibc/blob/8c6e9be5b94e9162146217772c2d3dceb2376c7f/options/internal/aarch64-include/mlibc/thread.hpp#L8-L13
https://github.com/managarm/mlibc/blob/8c6e9be5b94e9162146217772c2d3dceb2376c7f/options/rtld/generic/linker.cpp#L40